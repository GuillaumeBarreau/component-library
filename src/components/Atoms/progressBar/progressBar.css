/* progressBar */

.progressBar {
  width: 100%;
  transition: 0.5;
  background: var(--color-base-02);
  border-radius: 10px;
  box-shadow: 0px 4px 6px rgba(53, 69, 122, 0.4);
  position: relative;
  overflow: hidden;
}

.progressBar--size-small {
  height: 8px;
}

.progressBar--size-xsmall {
  height: 12px;
}

.progressBar--size-default {
  height: 16px;
}

.progressBar--size-medium {
  height: 20px;
}

.progressBar--content {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  transition: 1000ms;
  transition-timing-function: linear;
}

.progressBar--notice-primary {
  background: linear-gradient(
    var(--color-notice-primary--default),
    var(--color-notice-primary--dark)
  );
}

.progressBar--notice-secondary {
  background: linear-gradient(
    var(--color-notice-secondary--default),
    var(--color-notice-secondary--dark)
  );
}

.progressBar--notice-success {
  background: linear-gradient(
    var(--color-notice-success--default),
    var(--color-notice-success--dark)
  );
}

.progressBar--notice-warning {
  background: linear-gradient(
    var(--color-notice-warning--default),
    var(--color-notice-warning--dark)
  );
}

.progressBar--notice-error {
  background: linear-gradient(
    var(--color-notice-error--default),
    var(--color-notice-error--dark)
  );
}

.progressBar--notice-info {
  background: linear-gradient(
    var(--color-notice-info--default),
    var(--color-notice-info--dark)
  );
}
