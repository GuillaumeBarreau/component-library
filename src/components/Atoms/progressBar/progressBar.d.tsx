export interface ComponentProgressBarProps {
  className?: string;
  progessMin: number;
  progessMax: number;
  progress: number;
  size?: "xsmall" | "small" | "medium" | "default";
  notice?:
    | "success"
    | "info"
    | "warning"
    | "error"
    | "disabled"
    | "primary"
    | "secondary"
    | undefined;
}
