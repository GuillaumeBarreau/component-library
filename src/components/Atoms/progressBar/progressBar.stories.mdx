import { Meta, Story, Canvas, ArgsTable } from "@storybook/addon-docs";
import { ProgressBar } from "./progressBar";

<Meta 
  title="Components/Atoms/ProgressBar" 
  component={ProgressBar}
  argTypes={{
    notice: {
      control: "select",
      description: "Modify the background color to ProgressBar",
      options: [
        "default",
        "info",
        "warning",
        "success",
        "error",
      ],
    },
    progress: {
      description: "The progression of the progress bar in percentage (from 0 to 100) ",
      control: { type: 'range', min:1, max:100 },
    },
    className: {
      description: "Override a component's styles",
      control: "text",
    },
  }}
  args={{
    progress: 50,
  }}/>

# ProgressBar

The ProgressBar

## React Component

#### Import library

```
Import {Atoms} from '@bonjour-les-pixels/component-library';
```

#### Use ProgressBar component

```
<Atoms.ProgressBar {...props}></Atoms.ProgressBar>
```

## Playground

<Canvas>
  <Story name="ProgressBar">
    {args => {
      return <ProgressBar {...args}  />;
    }}
  </Story>
</Canvas>

### Props

<ArgsTable story="ProgressBar" />

### Notice

You can use the `notice` prop to modify the background color. This props can be declined in 5 ways. The style varies according to the level of importance of the action or the brand color in the page context.

<Canvas>
  <ProgressBar 
    progress="50"
  />
</Canvas>

<Canvas>
  <ProgressBar 
    progress="50"
    notice="info"
  />
</Canvas>

<Canvas>
  <ProgressBar 
    progress="50"
    notice="error"
  />
</Canvas>

<Canvas>
  <ProgressBar 
    progress="50"
    notice="success"
  />
</Canvas>

<Canvas>
  <ProgressBar 
    progress="50"
    notice="warning"
  />
</Canvas>

### ClassName

You can use the `className` prop to override a component's styles using custom classes.

```
  <Alert className="OVERRIDE_STYLE_CLASSNAME" />
```

### Size

You can use the size prop to modify the button size. You can set the value to `xsmal`, `small`, `default` or `medium`. 
The default size is default and therefore doesn't need to be specified.

<Canvas>
  <ProgressBar
      size="xsmall"
      progress="60"
    />
</Canvas>

<Canvas>
  <ProgressBar
      size="small"
       progress="60"
    />
</Canvas>

<Canvas>
  <ProgressBar
      size="default"
       progress="60"
    />
</Canvas>

<Canvas>
  <ProgressBar
      size="medium"
       progress="60"
    />
</Canvas>