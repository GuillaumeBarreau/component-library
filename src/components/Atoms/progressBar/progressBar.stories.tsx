import { ProgressBar } from "./progressBar";
import { Meta } from "@storybook/react/types-6-0";
import { ComponentProgressBarProps } from "./progressBar.d";

export default {
  title: "ProgressBar",
  component: ProgressBar,
  parameters: {
    a11y: {
      // optional selector which element to inspect
      element: "#root",
      // optional flag to prevent the automatic check
      manual: false,
    },
  },
} as Meta;

const Template: any = (
  args: JSX.IntrinsicAttributes & ComponentProgressBarProps
) => (
  <>
    <section style={{ display: "flex", justifyContent: "center" }}>
      <ProgressBar {...args} />
    </section>
  </>
);

export const ProgressBarDefault = Template.bind({});
ProgressBarDefault.args = {
  children: "progressBar",
};
