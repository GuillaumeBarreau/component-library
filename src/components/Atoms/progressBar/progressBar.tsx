import React from "react";
import classNames from "classnames";
import { ComponentProgressBarProps } from "./progressBar.d";
import "./progressBar.css";

export const ProgressBar: React.FC<ComponentProgressBarProps> = (props) => {
  const {
    className,
    notice = "primary",
    progessMin,
    progessMax,
    progress,
    size = "default",
  } = props;

  const prefixCls: string = "progressBar";
  const classes = classNames(
    prefixCls,
    {
      [`${prefixCls}--size-${size}`]: size,
    },
    className
  );

  return (
    <div className={classes}>
      <div
        className={`progressBar--content progressBar--notice-${notice}`}
        style={{ width: `${progress}%` }}
        aria-valuenow={progress}
        aria-valuemin={progessMin}
        aria-valuemax={progessMax}
      />
    </div>
  );
};
