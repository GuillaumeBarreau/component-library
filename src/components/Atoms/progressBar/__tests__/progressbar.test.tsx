import React from "react";
import { shallow } from "enzyme";
import { ProgressBar } from "../progressBar";
import { ComponentProgressBarProps } from "../progressBar.d";

describe("ProgressBar", () => {
  let defaultProps: JSX.IntrinsicAttributes & ComponentProgressBarProps;

  beforeEach(() => {
    defaultProps = {
      notice: "primary",
      progress: 50,
      size: "small",
      className: "override-class",
    };
  });

  it("renders correctly", () => {
    const wrapper = shallow(<ProgressBar {...defaultProps} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("mount correctly", () => {
    const wrapper = shallow(<ProgressBar {...defaultProps} />);
    expect(() => wrapper).not.toThrow();
  });

  it("Should contain defaultProps:children correctly", () => {
    const wrapper = shallow(<ProgressBar {...defaultProps} />);
    expect(() => wrapper).not.toThrow();
    expect(wrapper.html()).toContain("progressBar");
  });

  it("Should contain defaultProps:size correctly", () => {
    const wrapper = shallow(<ProgressBar {...defaultProps} />);
    expect(wrapper.html()).toContain("small");
  });

  it("Should contain defaultProps:className correctly", () => {
    const wrapper = shallow(<ProgressBar {...defaultProps} />);
    expect(wrapper.html()).toContain("override-class");
  });

  it("Should contain defaultProps:progessMax correctly", () => {
    const wrapper = shallow(<ProgressBar {...defaultProps} />);
    expect(wrapper.find(".progressBar--content").prop("aria-valuemax")).toEqual(
      100
    );
  });

  it("Should contain defaultProps:progress correctly", () => {
    const wrapper = shallow(<ProgressBar {...defaultProps} />);
    expect(wrapper.find(".progressBar--content").prop("aria-valuenow")).toEqual(
      50
    );
  });

  it("Should contain defaultProps:progessMin correctly", () => {
    const wrapper = shallow(<ProgressBar {...defaultProps} />);
    expect(wrapper.find(".progressBar--content").prop("aria-valuemin")).toEqual(
      0
    );
  });

  it("Should contain the witdh style value at 50% correctly", () => {
    const wrapper = shallow(<ProgressBar {...defaultProps} />);
    expect(wrapper.html()).toContain("width:50%");
  });
});
