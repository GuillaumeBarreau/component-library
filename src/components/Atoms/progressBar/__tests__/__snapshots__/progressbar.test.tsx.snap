// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders correctly 1`] = `
"<div className=\\"progressBar progressBar--size-small\\">
  <div className=\\"progressBar--content progressBar--notice-primary\\" style={{...}} aria-valuenow={50} aria-valuemin={0} aria-valuemax={100} />
</div>"
`;
